applications/views.py

```diff
@@ -127,11 +127,15 @@ def page(request, pk, key):
     except ValueError:
         raise Http404
 
+    unbound_form = page.get_unbound_data_form()
+
     if request.method == "GET":
-        form = page.get_unbound_data_form()
+        form = unbound_form
     else:
         form = page.get_bound_data_form(request.POST)
-
+        if form.has_changed():
+            page.instance.is_approved = False
+            page.instance.save()
```

Comment:
nit: could we keep a blank both before and after this block?  It can help to denote a separate block for some.